Rewrite callbacks

DenisKolodin opened the pull request:
The continuation of #211

vitiral commented on src/html.rs:
```diff
@@ -315,11 +316,11 @@ where
 }
 
 /// This handle keeps the reference to a detached scope to prevent memory leaks.
```
docs still call it `handle`.

Am I to understand that this helps with dropping memory when it is no longer used -- because javascript doesn't handle that automatically?

Adding a sentance explaining that would be _really_ useful.

DenisKolodin commented:
One night without sleep and this day has come:
**I've removed JS insertions!** (main loop and scheduler) :tada: :tada: :tada:
https://github.com/DenisKolodin/yew/pull/212/files#diff-b4aea3e418ccdb71239b96952d9cddb6L72

Why it's important:
* Performance will increase :rocket: 
* We could use `requestAnimationFrame` :muscle: 
* I could add `Pool`s of callbacks (with this thing I started :grinning: )

It's the hardest core improvement since 0.1. I replaced a scheduler and channels with cells under reference counters. But It's not lossless: child components were broken and I have to rewrite them. I still keep going.

vitiral commented on examples/two_apps/src/lib.rs:
```diff
@@ -52,13 +52,28 @@ where
+                    }
+                }
                 self.title = title;
+                drop(context);
```
isn't this called automatically?

DenisKolodin replied:
You are right, but I don't sure the optimizer won't return it immediately after initialization. But it's important to keep the lock till the end.

vitiral commented on src/virtual_dom/mod.rs:
```diff
@@ -78,7 +78,7 @@ pub trait VDiff {
     type Component: Component<Self::Context>;
 
     /// Remove itself from parent and return the next sibling.
```
nit: Remove -> Detach

DenisKolodin replied:
I haven't fixed the comments as not to waste time for them now. I want to fix it later.

DenisKolodin commented on examples/two_apps/src/lib.rs:
```diff
@@ -52,13 +52,28 @@ where
+                        self.activator.send_message(Msg::SetTitle("Pong Done".into()));
+                    }
+                    "Pong Done" => {
+                        self.activator.send_message(Msg::SetTitle("Ping Done".into()));
```
I think it's better to rename both :ok:

DenisKolodin commented:
bors r+
